src/main/java/fr/openent/competences/Utils.java

```diff
@@ -223,14 +223,20 @@ public void handle(Message<JsonObject> message) {
         }));
     }
 
+    public static Future<JsonArray> getGroupsEleve(EventBus eb, String studentId, String structureId) {
+        Promise<JsonArray> promise = Promise.promise();
+        getGroupsEleve(eb, studentId,structureId, FutureHelper.handler(promise,
+                "[Competences@Utils::getGroupsEleve] error during get sql request"));
+        return promise.future();
+    }
     /**
      * Get eleve group
      *
      * @param eb      EventBus
      * @param idEleve id Eleve
      * @param handler response eleve groups
      */
-
+    @Deprecated
```

Comment:
Tu dépends d une fonction déprecated : déplacer l'algo dans la fonction qui renvoie une future et l'appeler dans cette fonction

Reply:
Faire attention si la fonction est utilisée autre part, si elle déplace l'algo dans sa nouvelle fonction, code à réadapter dans les autres fonctions et peut-être pas dans le scope du ticket / risque de régression

Reply:
Il manque la java doc pour indiqué au dév la nouvelle fonction a utilisé
```suggestion
    /**
     * @deprecated Use {@link #getGroupsEleve(EventBus, String, String)}
     */
    @Deprecated
```

Reply:
Oui, finalement je laisse comme ça car sinon cela va impliquer plus de vérification et donc plus de temps, non prévu dans ce ticket.